test/ability_sap_sipper.c

```diff
@@ -0,0 +1,54 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Sap Sipper negates damage from Grass-type moves")
+{
+    GIVEN {
+        PLAYER(SPECIES_MARILL) { Ability(ABILITY_SAP_SIPPER); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_VINE_WHIP); }
+    } SCENE {
+        NONE_OF { HP_BAR(player); }
+    }
+}
+
+SINGLE_BATTLE_TEST("Sap Sipper negates effects from Grass-type moves")
+{
+    GIVEN {
+        PLAYER(SPECIES_MARILL) { Ability(ABILITY_SAP_SIPPER); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SPORE); }
+    } SCENE {
+        NONE_OF { ANIMATION(ANIM_TYPE_STATUS, B_ANIM_STATUS_SLP, player); }
+        NONE_OF { STATUS_ICON(player, sleep: TRUE); }
+    }
+}
+
+SINGLE_BATTLE_TEST("Sap Sipper increases Attack by one stage when hit by a Grass-type move")
+{
+    GIVEN {
+        PLAYER(SPECIES_MARILL) { Ability(ABILITY_SAP_SIPPER); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_VINE_WHIP); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_SAP_SIPPER);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
+        MESSAGE("Marill's Attack rose!");
```

Comment:
Let's add damage checks here.

Reply:
How would I check for a lack of damage (since it grants immunity)?

Reply:
You can say [`NONE_OF { HP_BAR(player); }`](<https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/battle/ability/ice_body.c#L12>)

Reply:
Isn't that effectively already covered by the very first test, though? I can obviously add it again, but I'm wondering if there's a point in having two tests in that case

Reply:
Oh, sorry, I haven't actually read the tests I just noticed the question :sweat_smile: Hopefully Edu can chime in

Reply:
> Isn't that effectively already covered by the very first test, though? I can obviously add it again, but I'm wondering if there's a point in having two tests in that case

You're right. Carry on.